Fluid/MemberAccessStrategy.cs

```diff
@@ -60,30 +59,31 @@ public IMemberAccessor GetAccessor(Type type, string name)
 
         public void Register(Type type, string name, IMemberAccessor getter)
         {
-            if (!_map.TryGetValue(type, out var typeMap))
+            lock (_map)
             {
-                typeMap = CreateTypeMap(type);
-            }
+                var localMap = _map;
 
-            typeMap[name] = getter;
-        }
+                // If it is already being used, we clone the structure
+                if (_initialized)
+                {
+                    localMap = new Dictionary<Type, Dictionary<string, IMemberAccessor>>(_map);
+                    foreach (var entry in localMap)
```

Comment:
somehow it would make more sense to me to iterate _map and intialize localMap with size instead of _map, now I guess localMap construction calculates hash codes and fills by buckets just to do it again in the next loop.